parlai/mturk/core/test/test_db_interactions.py

```diff
@@ -0,0 +1,547 @@
+        self.assertEqual(worker_2_data['worker_id'], worker_id_2)
+        self.assertEqual(worker_2_data['accepted'], 2)
+        self.assertEqual(worker_2_data['disconnected'], 0)
+        self.assertEqual(worker_2_data['completed'], 0)
+        self.assertEqual(worker_2_data['approved'], 0)
+        self.assertEqual(worker_2_data['rejected'], 0)
+
+        # Ensure all the assignments are marked as accepted
+        c.execute('SELECT COUNT(*) FROM assignments WHERE status = ?;',
+                  ('Accepted', ))
+        self.assertEqual(c.fetchone()[0], 3)
+
+        # Ensure non-existing assign is None
+        self.assertIsNone(db_logger.get_assignment_data('fake_id'))
+
+        # Check each of the assignments
+        assignment_1_data = db_logger.get_assignment_data(assignment_id_1)
+        assignment_2_data = db_logger.get_assignment_data(assignment_id_2)
+        assignment_3_data = db_logger.get_assignment_data(assignment_id_3)
+        self.assertEqual(assignment_1_data['assignment_id'], assignment_id_1)
+        self.assertEqual(assignment_1_data['status'], 'Accepted')
+        self.assertEqual(assignment_1_data['approve_time'], None)
+        self.assertEqual(assignment_1_data['worker_id'], worker_id_1)
+        self.assertEqual(assignment_1_data['hit_id'], HIT1['HIT']['HITId'])
+        self.assertEqual(assignment_2_data['assignment_id'], assignment_id_2)
+        self.assertEqual(assignment_2_data['status'], 'Accepted')
+        self.assertEqual(assignment_2_data['approve_time'], None)
+        self.assertEqual(assignment_2_data['worker_id'], worker_id_2)
+        self.assertEqual(assignment_2_data['hit_id'], HIT2['HIT']['HITId'])
+        self.assertEqual(assignment_3_data['assignment_id'], assignment_id_3)
+        self.assertEqual(assignment_3_data['status'], 'Accepted')
+        self.assertEqual(assignment_3_data['approve_time'], None)
+        self.assertEqual(assignment_3_data['worker_id'], worker_id_2)
+        self.assertEqual(assignment_3_data['hit_id'], HIT3['HIT']['HITId'])
+
+        # Ensure three pairings have been created, one for each assignment
+        c.execute('SELECT COUNT(*) FROM pairings')
+        self.assertEqual(c.fetchone()[0], 3)
+
+        # Ensure pairings are accurate
+        self.assertIsNone(db_logger.get_worker_assignment_pairing(
+            worker_id_1, assignment_id_3))
+
+        pair_1 = db_logger.get_worker_assignment_pairing(
+            worker_id_1, assignment_id_1)
+        pair_2 = db_logger.get_worker_assignment_pairing(
+            worker_id_2, assignment_id_2)
+        pair_3 = db_logger.get_worker_assignment_pairing(
+            worker_id_2, assignment_id_3)
+        for f in ['onboarding_start', 'onboarding_end', 'task_start',
+                  'task_end', 'conversation_id']:
+            for pair in [pair_1, pair_2, pair_3]:
+                self.assertIsNone(pair[f])
+        self.assertEqual(pair_1['status'], AssignState.STATUS_NONE)
```

Comment:
why are these `STATUS_NONE` here and not `Accepted`?

Reply:
Pairs track the assign state from the perspective of MTurkManager, where `STATUS_NONE` is essentially the valid state of `"Accepted"`. These should soon transition to `STATUS_ONBOARDING` or `STATUS_WAITING`, which is the more relevant info to track in a pairing.